crowbar_framework/app/models/nova_service.rb

```diff
@@ -184,7 +184,18 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
 
     neutron = ProposalObject.find_proposal("neutron",role.default_attributes["nova"]["neutron_instance"])
 
-    all_nodes.each do |n|
+    compute_nodes_for_network = []
+    role.override_attributes["nova"]["elements"].each do |role, nodes|
+      # only care about compute nodes
+      next unless role =~ /^nova-multi-compute-/
+      # vmware compute nodes do not need access to the networking
+      next if role == "nova-multi-compute-vmware"
```

Comment:
> Why wouldn't they need it? 

Honestly I am not even sure if IP allocations for something else than the admin network even work with the current state of the hyperv recipes. But otoh they shouldn't really hurt either. So I don't have any objection against merging this.